stdio/file.c

```diff
@@ -259,19 +256,21 @@ FILE *freopen(const char *pathname, const char *mode, FILE *stream)
 {
 	int m;
 
-	if (mode == NULL || stream == NULL)
+	if (stream == NULL) {
+		errno = EBADF;
```

Comment:
_EBADF: The mode with which the file descriptor underlying the stream was opened does not support the requested mode when pathname is a null pointer._

This isn't really good return value in this case. Case of `stream` being `NULL` is not handled by the standard and most implementations don't check for `NULL` - they crash instead. This approach have it's benefits - user should check `stream` for `NULL` (after `fopen`), so `NULL` shouldn't be passed here ever. So this check is redundant and causes (very slight) performance hit.

IMHO - `EINVAL` or drop this check altogether.

Reply:
But how about:
>The freopen() function shall fail if:
[EBADF] The file descriptor underlying the stream is not a valid file descriptor when pathname is a null pointer. 

However I will change it to `EINVAL` as suggested. We need this check as then there is `fflush()` which will not throw an error when passed `NULL` an will flush `stdout`. So we might need this check to not flush stdout for no reason if function fails.